### ISSUE TITLE
When upgrading from angular1, updating a $watched value from a setTimeout ends up outside ngZone

### ISSUE THREAD

**Issue by kvantetore** [angular/angular#7252]
Consider the following code

```
        var x = "hello";

        setTimeout(() => {
            x = "world!";
        }, 100);

        $scope.$watch(() => x, () => {
            console.log("Got x == " + x + " in zone " + window.zone.$id);
        });
```

I would expect that the watch would always be called inside ngZone. However, after fixing #6385, the above watch will report being run in zone 1 and thus breaking further change detection.

Changing upgade_adapter back to ngZone.run(() => rootScope.$apply()) fixes the issue.

**Comment by huyph** [comment]
I actually have the same issue in my project and it is kind of hard to replicate with a plunker. 

Perhaps we should submit a PR to revert that fix, to open up a discussion on this?

**Comment by gkalpak** [comment]
The fix for #6385 has been reverted, so this should work as expected again.
